solr/core/src/java/org/apache/solr/search/SortSpecParsing.java

```diff
@@ -220,6 +185,40 @@ private static SortSpec parseSortSpecImpl(
     return new SortSpec(s, fields);
   }
 
+  static ValueSource parseValueSource(QParser parser, StrParser sp, int start) throws SyntaxError {
```

Comment:
de-duplicated this code, and now using `fparser.parseAsValueSource`.  Only 2 callers within Solr so using package scope / internal.